test/images/image-util.sh

```diff
@@ -21,63 +21,63 @@ set -o pipefail
 TASK=$1
 IMAGE=$2
 
-KUBE_ROOT="$(cd "$(dirname "${BASH_SOURCE}")/../.." && pwd -P)"
+KUBE_ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/../.." && pwd -P)"
 source "${KUBE_ROOT}/hack/lib/util.sh"
 
 # Mapping of go ARCH to actual architectures shipped part of multiarch/qemu-user-static project
 declare -A QEMUARCHS=( ["amd64"]="x86_64" ["arm"]="arm" ["arm64"]="aarch64" ["ppc64le"]="ppc64le" ["s390x"]="s390x" )
 
 # Returns list of all supported architectures from BASEIMAGE file
 listArchs() {
-  cut -d "=" -f 1 ${IMAGE}/BASEIMAGE
+  cut -d "=" -f 1 "${IMAGE}/BASEIMAGE"
 }
 
 # Returns baseimage need to used in Dockerfile for any given architecture
 getBaseImage() {
   arch=$1
-  echo $(grep "${arch}=" BASEIMAGE | cut -d= -f2)
+  grep "${arch}=" BASEIMAGE | cut -d= -f2
 }
 
 # This function will build test image for all the architectures
 # mentioned in BASEIMAGE file. In the absence of BASEIMAGE file,
 # it will build for all the supported arch list - amd64, arm,
 # arm64, ppc64le, s390x
 build() {
-  if [[ -f ${IMAGE}/BASEIMAGE ]]; then
-    archs=$(listArchs)
+  if [[ -f "${IMAGE}/BASEIMAGE" ]]; then
+    while IFS='' read -r arch; do archs+=("$arch"); done < <(listArchs)
```

Comment:
There was intermixed usage of arrays and strings. For this I opted to go the array route.